pkg/schema/type.go

```diff
@@ -177,11 +179,18 @@ func (m *ScalarType) CheckType(node yamlmeta.TypeWithValues) (chk yamlmeta.TypeC
 			chk.Violations = append(chk.Violations,
 				NewMismatchedTypeError(node, m))
 		}
-	case int:
-		if _, ok := m.Value.(int); !ok {
+	case float64:
+		if _, ok := m.Value.(float64); !ok {
 			chk.Violations = append(chk.Violations,
 				NewMismatchedTypeError(node, m))
 		}
+	case int:
+		if _, ok := m.Value.(int); !ok {
+			if _, ok = m.Value.(float64); !ok {
```

Comment:
is this code saying that we accept float for type integer fields?

Reply:
The opposite, that we accept integers for float fields.

It looks a little weird, because we're switching on the type of the thing to check, not the schema. So we're saying if we got an int, make sure the schema is an int or a float.